tests/sender/send_data.py

```diff
@@ -315,3 +322,228 @@ def test_sender_as_handler_static(self):
+                                        key=self.key,
+                                        cert=self.cert,
+                                        chain=self.chain,
+                                        check_hostname=False,
+                                        verify_mode=CERT_NONE,
+                                        verify_config=True)
+        result = Sender.check_config_certificate_chain(engine_config)
+        self.assertEqual(result, True)
+
+    def test_config_cert_chain_incompatible_case(self):
+        """
+        Test that verifies that an incompatible
+        certificate with a chain raises an exception.
+        """
+
+        engine_config = SenderConfigSSL(address=(self.server, self.port),
+                                        key=self.key,
+                                        cert=self.cert,
+                                        chain=self.chain,
+                                        check_hostname=False,
+                                        verify_mode=CERT_NONE,
+                                        verify_config=False)
+        incompatible_cert = \
+            "{!s}/local_certs/keys/server/server_cert.pem" \
+            .format(os.path.dirname(os.path.abspath(__file__)))
+        engine_config.chain = incompatible_cert
+        with self.assertRaises(DevoSenderException) as result:
+            Sender.check_config_certificate_chain(engine_config)
+
+        self.assertEqual(
+            "Error in config, the chain: " + engine_config.chain +
+            " is not compatible with the certificate: " +
+            engine_config.cert,
+            str(result.exception))
+
+    def test_config_cert_address_standard_case(self):
+        """
+        Test that verifies that a compatible certificate
+        and address are detected.
+        """
+        engine_config = SenderConfigSSL(address=(self.server, self.port),
+                                        key=self.key,
+                                        cert=self.cert,
+                                        chain=self.chain,
+                                        check_hostname=False,
+                                        verify_mode=CERT_NONE,
+                                        verify_config=True)
+
+        def fake_get_peer_cert_chain():
+            chain = open(engine_config.chain, "rb").read()
+            chain_certs = []
+            for _ca in pem.parse(chain):
+                chain_certs.append(
+                    crypto.load_certificate(
+                        crypto.FILETYPE_PEM, str(_ca)))
+                chain_certs.append(
+                    crypto.load_certificate(
+                        crypto.FILETYPE_PEM, str(_ca)))
+            return chain_certs
+
+        with mock.patch.object(
+                SSL.Connection, 'get_peer_cert_chain',
+                mock.MagicMock(side_effect=fake_get_peer_cert_chain))\
+                as fake_get_peer_cert_chain_mock:
```

Comment:
*issue(blocking):* This is way too complicated for a test case. The mocking should occur for code that you own unless for specific reasons, so patching a method from `SSL.Connection` is, unfortunately, a bad practice. Plus, the operation to patch should be dead simple, because otherwise, we should be creating additional tests for `fake_get_peer_cert_chain` to prove it is working as expected. You should patch specific operations from your own code, and have patches that will just skip the tedious operative you want to avoid in the simplest form possible.

Reply:
The test is trying to demonstrate that a certificate is valid for an URL, in this test case the URL is `localhost`, that's why I had to mock the `get_peer_cert_chain` to get a compatible certificate from `localhost`, I could get rid of the `fake_get_peer_cert_chain` if I find a way to get certificates from `localhost` or to use another URL besides `localhost` and have valid certificates for that URL, I will take any suggestions.

Reply:
If this is the only way, I would at least try to simplify the `fake_get_peer_cert_chain` function and extract it somewhere else in the script (to avoid having a nested function, basically).